packages/execution-environments/src/common/endowments/interval.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Creates a pair of `setInterval` and `clearInterval` functions attenuated such
+ * that:
+ * - `setInterval` throws if its "handler" parameter is not a function.
+ * - `clearInterval` only clears timeouts created by its sibling `setInterval`,
+ *   or else no-ops.
+ *
+ * @returns An object with the attenuated `setInterval` and `clearInterval`
+ * functions.
+ */
+const createInterval = () => {
+  const registeredIntervals = new Set<number>();
+
+  const _setInterval = (handler: TimerHandler, timeout?: number): number => {
+    if (typeof handler !== 'function') {
+      throw new Error(
+        `The interval handler must be a function. Received: ${typeof handler}`,
+      );
+    }
+
+    const handleWrapper: { handle: number } = { handle: NaN };
+    // @todo Should we return a number that we decide for NodeJS?
```

Comment:
handles could be just `unknown` (as well as `registeredIntervals = new Set<unknown>()` and everywhere else a handle is expected) and that would work for both browser and NodeJS environments

Reply:
Might as well make the same change to `setTimeout`